### DISCUSSION TITLE
[Angular] Once an error in the effects chain occurs, no effects are triggered anymore

### DISCUSSION THREAD

**Discussion by edskeizer** [salesforce/akita#693] [Q&A] [answered]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior
If an error occurs in the chain of an effect, all effects dispatched afterwards are ignored.

## Expected behavior
I would expect subsequent effects to be handled normally.

## Minimal reproduction of the problem with instructions
See this Stackblitz with the issue. I made an example api call in TodoService that throws an error when adding a second todo. A third todo should be created, but is ignored.
https://stackblitz.com/edit/akita-todos-app-edctvf?embed=1&file=src/app/todos/state/todos.effects.ts

## What is the motivation / use case for changing the behavior?
When a todo api call fails (for some reason), the next addition should just work. This is not the case now: all actions dispatched afterwards are ignored.

## Environment

<pre><code>
Angular version: 12.0.3

Browser:
- [x] Chrome (desktop) version 91
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

</code></pre>

**Comment by NetanelBasal** [comment]
@yackinn

**Comment by yackinn** [comment]
I'll look into it.

**Comment by yackinn** [comment, marked as answer]
@edskeizer 
You need to catch the error on your inner observable issued by the switchmap operator.
If you catch the the error on the obserable you complete the observable stream.

```ts
createTodo$ = this.actions$.pipe(
	ofType(createTodoRequestAction),
	switchMap((payload: { todoName: string }) =>
	  this.todosService.add$(payload.todoName).pipe(
	    map(todo => createTodoSuccessAction(todo)),	
        catchError(error => of(createTodoErrorAction(error)))
	  )
  	),
);
```
